crates/proto/src/op/message.rs

```diff
@@ -682,6 +684,39 @@ impl Message {
 
         Ok(())
     }
+
+    #[allow(clippy::type_complexity)]
+    /// Consumes `Message` and returns into components
+    pub fn into_parts(
+        self,
+    ) -> (
+        Header,
```

Comment:
So, I had this thought that we might want to have a type with these fields public so that it can be destructured with proper name bindings to the individual fields...

My concern with this as it is, is that it could be fairly error prone in terms of accidentally binding the wrong field.

We *could* make Message itself have public fields, but there is a lot of logic around how and when some of those are generated I think, where it might be dangerous to offer that on the base type itself. What do you think about just adding a simple type that is just all of these fields and return that instead?

Reply:
So this is a type with public fields such that you can destructure, the fields are just not named. I think the fear of binding to the wrong field is greatly reduced because the types won't match even if you do, however I don't feel strongly and can go either way. I think keeping the `Message` type with private fields is probably the right call, so if you want to have named fields here a new type would be required.

tangent: I had a thought the other day that perhaps it would be nice to have a `MessageBuilder` builder-pattern style type for constructing `Messages` as opposed to constructing them directly.... something to think about

Reply:
It just occurred to me that the `Vec<Record>` fields all have the same type so perhaps we do want to name them...

Reply:
Actually, yeah, I really like the idea of adding a MessageBuilder. That could be passed around instead of Message. That's probably a bigger change then you want in this PR though.

The specific fields I'm concerned about are the Answers, Authorities, and Additionals. Those are all `Vec<Record>` which is where my specific concern comes from. I agree that the others are strongly typed.

Reply:
I'll leave `MessageBuilder` for a future PR, sure. The type alias might be nice, but I don't think it would prevent you from mistakenly passing in `Vec<Record>` because I'm *pretty* sure they would all be equivalent.

```
type Additionals = Vec<Record>;
type Answers = Vec<Record>;
type Authorities = Vec<Record>;

fn take_answers(ans: Answers) { .. }
```
`take_answers` could be called with `Additoinals` / `Authorities` also because they all alias the same type. If we wanted to make it a compiler error we'd need a newtype wrapper:

```
struct Additionals(Vec<Record>);
```

Given all that... is there any approach that you like more?

Reply:
How about a new type, `MessageParts` for the return with all fields, `pub`, sound good to you?